src/connection-manager/agreement.ts

```diff
@@ -178,6 +197,10 @@ class Agreement {
         last_ledger_time: new Date(),
       }
 
+      if (server_version) {
```

Comment:
nit: prefer `server_version != null`

Reply:
That is equal to what is there.  Did you instead mean `server_version !== null`?

Reply:
I prefer `!=`. Willing to hear other opinions, though.

`server_version != null` will not enter the `if ()` statement if we ever hit a case where `server_version` is `undefined`. 

On the off chance we hit an `undefined` server_version, we don't want to enter that if statement.

The following script:
```js
const foo = undefined;

if (foo !== null)
  console.log(`hit case !== with ${foo}`)

if (foo != null)
  console.log(`hit case != null with ${foo}`)
```

Outputs: 
```
hit case !== with undefined
```

Worth noting, I don't think this actually will change anything with the runtime behavior of this function. I _think_ `if (...)` evaluates a very similar way.

Reply:
Also worth noting that I no longer own this repo, so if there's a style guide y'all use you're free to disregard my opinions :)

Reply:
Correct.  Most linting configs frown on `!=`.  I have not looked into this repos rules yet.

Reply:
I also prefer `server_version != null` - specifically with `null` you want to use just `==` or `!=` (I believe most linters respect that, we definitely have a bunch of those in xrpl.js). That way, it also catches `undefined` and similar null-like values.

Reply:
Since `server_version = isPreceedingFlagLedger(validation.ledger_index) && decodeServerVersion(validation.server_version)`, it can also take a value of `false` when `isPreceedingFlagLedger(validation.ledger_index)` return `false`. So should it be `server_version != null && server_version !== false`?

Reply:
If that's the case, I'm okay with `if (server_version)`